src/lib/snack-bar/snack-bar-container.scss

```diff
@@ -33,6 +33,12 @@ $mat-snack-bar-spacing-margin: 24px !default;
     transform: translateY(-100%);
   }
 
+  &.mobile {
```

Comment:
A few notes:

* `.mobile` seems like something that could clash with user styles. What about calling it `mat-snackbar-mobile`? 
* Renaming the class will let you avoid the nesting.
* Is there a particular reason for going with viewport units rather than percent? I have a vague memory about `100vh` causing content to go behind the scrollbar on Windows.

Reply:
Yep, definitely forgot to rename the class after I got it working.

Because the overlay container does not establish itself as the entire width of the viewport, `width: 100%` turns into the width of the child based on content, so we use `100vw` .

Reply:
I see. That seems to go behind the scrollbar though. Here's a `position: fixed` div with a width of `100vw`:

![https://i.imgur.com/NJUaPFA.png](https://i.imgur.com/NJUaPFA.png)

You might be able to work around it by adding a class to the overlay panel via the `panelClass` and styling it from there.

Reply:
The issue that we run into with changing the width of the overlay container, is that the snackbar will always stay in its most expanded state (as limited by `max-width`)

I will have to continue to investigate